pkg/gcp/client/storage.go

```diff
@@ -56,21 +57,124 @@ func NewStorageClientFromSecretRef(ctx context.Context, c client.Client, secretR
 	return NewStorageClient(ctx, serviceAccount)
 }
 
-func (s *storageClient) CreateBucketIfNotExists(ctx context.Context, bucketName, region string) error {
-	if err := s.client.Bucket(bucketName).Create(ctx, s.serviceAccount.ProjectID, &storage.BucketAttrs{
-		Name:     bucketName,
-		Location: region,
+// CreateOrUpdateBucket ensures that a GCS bucket with the specified name exists in the given region,
+// applying the provided configuration. If the bucket does not exist, it is created. If the bucket
+// already exists and the immutability settings differ from the desired configuration, the retention
+// policy is updated accordingly.
+//
+// The config parameter can include immutability settings, such as a retention period. If a retention
+// policy is specified and not already locked, it will be locked to prevent further changes.
+//
+// Returns an error if the bucket creation or update fails.
+func (s *storageClient) CreateOrUpdateBucket(ctx context.Context, bucketName, region string, config *apisgcp.BackupBucketConfig) error {
```

Comment:
Ideally the client would not know about the BackupBucketConfig. This should be done on the actuator side rather than the gcp client abstraction level

Reply:
Fixed as part of 7e880c0dc3f7139e856be5b209206162d2debc70